spec/api-app-spec.js

```diff
@@ -805,33 +805,57 @@ describe('app module', () => {
   })
 
   describe('getGPUInfo() API', () => {
+    const appPath = path.join(__dirname, 'fixtures', 'api', 'gpu-info.js')
+
     it('succeeds with basic GPUInfo', (done) => {
-      app.getGPUInfo('basic').then((gpuInfo) => {
-        // Devices information is always present in the available info
-        expect(gpuInfo.gpuDevice).to.be.an('array')
-        expect(gpuInfo.gpuDevice.length).to.be.greaterThan(0)
+      const appProcess = ChildProcess.spawn(remote.process.execPath, [appPath, 'basic'])
+      let gpuInfoData = ''
+      appProcess.stdout.on('data', (data) => {
+        gpuInfoData += data
+      })
+      appProcess.stdout.on('end', () => {
+        const gpuInfo = JSON.parse(gpuInfoData.toString())
+        // Devices information is always present in the available info.
+        expect(gpuInfo).to.have.own.property('gpuDevice')
+          .that.is.an('array')
+          .and.is.not.empty()
+
         const device = gpuInfo.gpuDevice[0]
         expect(device).to.be.an('object')
-        expect(device)
-          .to.have.property('deviceId')
+          .and.to.have.property('deviceId')
           .that.is.a('number')
           .not.lessThan(0)
         done()
       })
     })
 
     it('succeeds with complete GPUInfo', (done) => {
-      app.getGPUInfo('complete').then((completeInfo) => {
-        // Driver version is present in the complete info
-        expect(completeInfo.auxAttributes.glVersion).to.be.a('string').that.has.length.greaterThan(0)
+      const appProcess = ChildProcess.spawn(remote.process.execPath, [appPath, 'basic'])
+
+      let gpuInfoData = ''
+      appProcess.stdout.on('data', (data) => {
+        gpuInfoData += data
+      })
+      appProcess.stdout.on('end', () => {
+        console.log(gpuInfoData)
+        const completeInfo = JSON.parse(gpuInfoData.toString())
```

Comment:
Maybe it would make sense to move new process creation and data retrieving logic to a separate function?
So here you could just `const info = await getGpuInfo('basic')`?